lib/impress.security.js

```diff
@@ -157,99 +155,96 @@ impress.security.mixin = (application) => {
+      .one()
+      .fetch((err, user) => {
+        if (err) return callback(err);
+        callback(null, application.security.user(user));
+      });
   };
 
   application.security.createSession = (
     // Create session in database
     session, // session instance
     callback // function
   ) => {
-    if (application.security.hasDb()) {
-      session.category = 'Sessions';
-      application.databases.security.create(session, callback);
-    } else callback();
+    if (!application.security.hasDb()) return callback();
+    session.category = 'Sessions';
+    application.databases.security.create(session, callback);
   };
 
   application.security.updateSession = (
     // Update session in database
     session, // session instance
     callback // function
   ) => {
-    if (application.security.hasDb()) {
-      session.category = 'Sessions';
-      application.databases.security.update(session, callback);
-    } else callback();
+    if (!application.security.hasDb()) return callback();
+    session.category = 'Sessions';
+    application.databases.security.update(session, callback);
   };
 
   application.security.readSession = (
     // Restore session from database if available
     sid, // session id
     callback // function(err, session)
   ) => {
-    if (application.security.hasDb()) {
-      application.databases.security
-        .select({ category: 'Sessions', sid })
-        .one()
-        .fetch((err, session) => {
-          if (err) callback(new Error('Session not found'));
-          else {
-            const login = session.login;
-            delete session._id;
-            session = application.security.session(session);
-            application.sessions.set(sid, session);
-            if (login) {
-              const user = application.users.get(login);
-              if (user) {
-                user.sessions.push(sid);
-                callback(null, session);
-              } else {
-                application.security.getUser(login, (err, item) => {
-                  if (item) {
-                    const user = application.security.user(item);
-                    user.sessions.push(sid);
-                    application.users.set(login, user);
-                  } else {
-                    application.sessions.delete(sid.login);
-                  }
-                  callback(null, session);
-                });
-              }
-            } else callback(null, session);
-          }
-        });
-    } else callback(new Error('No database for security subsystem found'));
+    if (!application.security.hasDb()) {
+      return callback(new Error('No database for security subsystem found'));
+    }
+    application.databases.security
+      .select({ category: 'Sessions', sid })
+      .one()
+      .fetch((err, session) => {
+        if (err) {
+          return callback(new Error('Session not found'));
+        }
+        const login = session.login;
+        delete session._id;
```

Comment:
Do we really need to delete `_id`? I'd rather have an extra pointer than change an object form.